### ISSUE TITLE
GAM class prediction bug

### ISSUE THREAD

**Issue by topepo** [tidymodels/parsnip#541]
``` r
library(tidymodels)
#> Registered S3 method overwritten by 'tune':
#>   method                   from   
#>   required_pkgs.model_spec parsnip
tidymodels_prefer()

data("parabolic")

gam_spec <- gen_additive_mod() %>% set_mode("classification")
vars <- workflow_variables(outcomes = c(class), predictors = c(starts_with("X")))

wflow <- 
   workflow() %>% 
   add_model(gam_spec, formula = class ~ s(X1) + s(X2)) %>% 
   add_variables(variables = vars)

fit <- fit(wflow, parabolic)
predict(fit, parabolic[1:3, -3])
#> # A tibble: 3 x 1
#>   .pred_class
#>   <fct>      
#> 1 Class1     
#> 2 Class2     
#> 3 Class2
```

<sup>Created on 2021-07-30 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

The issue is that `mgcv` returns a 1D _array_ of class probabilities. 

```
#   Browse[2]> 
#   debug: if (!is.null(object$spec$method$pred$class$post)) {
#       res <- object$spec$method$pred$class$post(res, object)
#   }
#   Browse[2]> res
#            1          2          3 
#   0.05420371 0.97374897 0.81707404 
#   Browse[2]> class(res)
#   [1] "array"
```

**Comment by DavisVaughan** [comment]
should this be closed then?

**Comment by juliasilge** [comment]
Yep!

Closed by #542
